src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs

```diff
@@ -46,12 +55,122 @@ public ApplicationExtract(ILogger<ApplicationExtract> log, ApplyDataContext appl
+
+                var submittedAnswersCompaniesHousePsCs = ExtractOrganisationPersonnel(answers, application.OrganisationId, QuestionIdCompaniesHousePSCs, PersonnelType.PersonWithSignificantControl);
+                organisationPersonnel.AddRange(submittedAnswersCompaniesHousePsCs);
+
+                var submittedAnswersCharityTrustees = ExtractOrganisationPersonnel(answers, application.OrganisationId, QuestionIdCharityTrustees, PersonnelType.CharityTrustee);
+                organisationPersonnel.AddRange(submittedAnswersCharityTrustees);
+
+                var submittedAnswersSoleTrade = ExtractOrganisationPersonnel(answers, application.OrganisationId, QuestionIdSoleTrade, PersonnelType.PersonInControl);
+                organisationPersonnel.AddRange(submittedAnswersSoleTrade);
+
+                var submittedAnswersPartnership = ExtractOrganisationPersonnel(answers, application.OrganisationId, QuestionIdPartnership, PersonnelType.PersonInControl);
+                organisationPersonnel.AddRange(submittedAnswersPartnership);
+
+                _applyDataContext.OrganisationPersonnel.AddRange(organisationPersonnel);
+
+                await _applyDataContext.SaveChangesAsync();
+
+                _logger.LogInformation($"Organisation Personnel successfully load for application {applicationId}");
+            }
+
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Unable to load Organisation Personnel for Application: {applicationId}");
             }
         }
 
+        private static List<OrganisationPersonnel> ExtractOrganisationPersonnel(List<SubmittedApplicationAnswer> answers, Guid organisationId, string questionId, PersonnelType personnelType)
+        {
+            var submittedAnswersOrganisationPersonnel = answers.Where(answer => answer.QuestionId == questionId).GroupBy(a => a.RowNumber).ToList();
+            var organisationPersonnel = new List<OrganisationPersonnel>();
+
+            if (personnelType == PersonnelType.PersonInControl && questionId == QuestionIdSoleTrade)
+            {
+                var submittedAnswersOrganisationNameSoleTrade = answers.Where(answer => answer.QuestionId == QuestionIdOrganisationNameSoleTrade);
+                if (submittedAnswersOrganisationPersonnel.Count > 0)
+                {
+                    foreach (var person in submittedAnswersOrganisationPersonnel)
+                    {
+                        var orgPersonnel = new OrganisationPersonnel
+                        {
+                            OrganisationId = organisationId,
+                            PersonnelType = (int)personnelType,
+                            Name = submittedAnswersOrganisationNameSoleTrade.FirstOrDefault().Answer
+                        };
+                        foreach (SubmittedApplicationAnswer record in person)
+                        {
+                            var dobArray = record.Answer.Split(",");
```

Comment:
I think you need a bit of protective code to check this isn't null, and it contains 2 and 2 only values in the array